test/models/drive_test.dart

```diff
@@ -0,0 +1,1175 @@
+          seats: 1);
+      expect(drive.isRidePossible(ride), true);
+    });
+    test('returns false if ride not able to fit in with a drive with many riders', () async {
+      final Drive drive = DriveFactory().generateFake(
+        startTime: DateTime(2022, 2, 2, 14),
+        endTime: DateTime(2022, 2, 2, 16),
+        seats: 4,
+        rides: <Ride>[
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14),
+              endTime: DateTime(2022, 2, 2, 15),
+              status: RideStatus.approved,
+              seats: 1),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14),
+              endTime: DateTime(2022, 2, 2, 14, 20),
+              status: RideStatus.approved,
+              seats: 3),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 20),
+              endTime: DateTime(2022, 2, 2, 15),
+              status: RideStatus.approved,
+              seats: 2),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 40),
+              endTime: DateTime(2022, 2, 2, 14, 50),
+              status: RideStatus.approved,
+              seats: 1),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 15),
+              endTime: DateTime(2022, 2, 2, 15, 30),
+              status: RideStatus.approved,
+              seats: 2),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 15),
+              endTime: DateTime(2022, 2, 2, 15, 20),
+              status: RideStatus.approved,
+              seats: 1),
+        ],
+      );
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime(2022, 2, 2, 14, 15),
+          endTime: DateTime(2022, 2, 2, 15, 10),
+          status: RideStatus.approved,
+          seats: 1);
+      expect(drive.isRidePossible(ride), false);
+    });
+  });
+  group('Drive.cancel', () {
+    test('cancelled is being changed from false to true', () async {
+      final Drive drive = DriveFactory().generateFake(cancelled: false, createDependencies: false);
+      when.call(driveProcessor.processUrl(any)).thenReturn(drive.toString());
+      drive.cancel();
+      expect(drive.cancelled, true);
+    });
+  });
+  group('Drive.toString', () {
+    test('returns a string representation of the drive', () async {
+      final Drive drive = DriveFactory().generateFake(
+        id: 1,
+        start: 'start',
+        startTime: DateTime.parse('2022-02-02T00:00:00.000Z'),
+        end: 'end',
+        endTime: DateTime.parse('2023-03-03T00:00:00.000Z'),
+        driverId: 5,
+        createDependencies: false,
+      );
+      expect(
+        drive.toString(),
+        'Drive{id: 1, from: start at 2022-02-02 00:00:00.000Z, to: end at 2023-03-03 00:00:00.000Z, by: 5}',
+      );
+    });
+  });
+  //soll equals auch auf rides prüfen und auf driver?
+  group('Drive.equals', () {
```

Comment:
Do we want to write these tests for every model... I feel like that's just a few hundred lines of code that is not particularly insightful

Reply:
Should I reduce the tests for equal? Or should we just leave them there since they already exist

Reply:
Let's leave them here, they don't hurt right now